python/packages/kagent-adk/src/kagent/adk/_agent_executor.py

```diff
@@ -273,5 +275,80 @@ async def _prepare_session(self, context: RequestContext, run_args: dict[str, An
             )
             # Update run_args with the new session_id
             run_args["session_id"] = session.id
-
+        if context.call_context:
+            if session.state is None:
+                session.state = {}
         return session
+
+
+def wrap_session(session_service, token: str):
```

Comment:
Nit: can we just define this class normally either in this file or a different one, rather than doing all of the imports and declaration in line? This feels more confusing personally